File: c#/src/Docmost.Sdk/Models/CursorPaginationMeta.cs
namespace Docmost.Sdk.Models;

public class CursorPaginationMeta
{
    [System.Text.Json.Serialization.JsonPropertyName("limit")]
    public int? Limit { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("hasNextPage")]
    public bool? HasNextPage { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("hasPrevPage")]
    public bool? HasPrevPage { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("nextCursor")]
    public string? NextCursor { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("prevCursor")]
    public string? PrevCursor { get; set; }

}

File: go/docmost/models/cursor_pagination_meta.go
package models

// CursorPaginationMeta from OpenAPI.
type CursorPaginationMeta struct {
	Limit *int `json:"limit,omitempty"`
	HasNextPage *bool `json:"hasNextPage,omitempty"`
	HasPrevPage *bool `json:"hasPrevPage,omitempty"`
	NextCursor *string `json:"nextCursor,omitempty"`
	PrevCursor *string `json:"prevCursor,omitempty"`
}
